collect_emails.py

```diff
@@ -76,27 +77,34 @@ def save_plain_text_content(email_bytes, email_id):
     save_chunks_to_vault(chunks)
     return text_content
 
-def search_and_process_emails(imap_client, email_source, search_keyword, start_date, end_date):
+def search_and_process_emails(imap_client, email_source, search_keyword, start_date, end_date, batch_size=100):
     search_criteria = 'ALL'
     if start_date and end_date:
         search_criteria = f'(SINCE "{start_date}" BEFORE "{end_date}")'
     if search_keyword:
-        search_criteria += f' BODY "{search_keyword}"'  # Ensure the correct combination of conditions
+        search_criteria += f' BODY "{search_keyword}"'

```

Comment:
this line did not meaningfully change. I'd ask that you avoid the number of needless edits, as it complicates review.